### PR TITLE
Allow `query` on local backend stack snapshots

### PR DESCRIPTION
**Pull request by hausdorff**

### CODE REVIEW

File: pkg/backend/filestate/backend.go [hunk 1]
```diff
@@ -550,11 +551,58 @@ func (b *localBackend) apply(
 
 // query executes a query program against the resource outputs of a locally hosted stack.
 func (b *localBackend) query(ctx context.Context, op backend.QueryOperation,
-	events chan<- engine.Event) result.Result {
+	callerEventsOpt chan<- engine.Event) result.Result {
```
Comment (hausdorff) [inline pkg/backend/filestate/backend.go]: All of this code could be shared with the cloud backend, but I'm not entirely sure where to put it.

**Comment by lblackstone** [comment]
@pgavlin Can you take a look at this PR soon? k8s provider CI has been failing for several days, and needs this change to fix. (It's pointed at `pulumi/pulumi:dev`, which is a separate issue that might be worth reconsidering)

File: pkg/backend/filestate/state.go [hunk 2]
```diff
@@ -49,6 +49,19 @@ const DisableCheckpointBackupsEnvVar = "PULUMI_DISABLE_CHECKPOINT_BACKUPS"
 // be used as a last resort when a command absolutely must be run.
 var DisableIntegrityChecking bool
 
+type cloudQuery struct {
```
Comment (pgavlin) [inline pkg/backend/filestate/state.go]:
```suggestion
type localQuery struct {
```

File: pkg/backend/filestate/backend.go (hunk 1 above)
Comment (pgavlin) [reply pkg/backend/filestate/backend.go]: `pkg/backend/query`, perhaps?

File: pkg/backend/filestate/state.go (hunk 2 above)
Comment (hausdorff) [reply pkg/backend/filestate/state.go]: Heh. This actually is meant to reflect that it is part of the `CloudQuery` SDK! But, ok. `localQuery` it is.

File: pkg/backend/filestate/backend.go (hunk 1 above)
Comment (hausdorff) [reply pkg/backend/filestate/backend.go]: it seems like a mild layer violation to put it there, but that's fine I guess.

**Pull request merged**